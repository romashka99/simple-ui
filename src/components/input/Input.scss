@import '../../styles/colors.scss';
@import '../../styles/main.scss';
@import '../../styles/style.scss';

.input {
	display: flex;
	position: relative;

	& .input--label {
		position: absolute;
		top: -16px;
		font-size: 12px;
		line-height: 16px;
	}

	& .input--inner {
		padding: 12px 8px;
		border: $border;
		border-radius: $border-radius;
		background-color: $color-white;
		color: $color-grey;
		height: 48px;
		line-height: 24px;
		font-size: 16px;
		box-sizing: border-box;

		&::placeholder {
			color: $color-grey;
		}

		&:disabled {
			border-color: $color-mid-light-grey;
			color: $color-mid-light-grey;
			cursor: not-allowed;

			&::placeholder {
				color: $color-mid-light-grey;
			}

			.input--label + & {
				color: $color-mid-light-grey;
			}
		}

		&:not(:disabled) {
			&:hover {
			}
			&:active,
			&:focus {
			}
		}
	}
}
