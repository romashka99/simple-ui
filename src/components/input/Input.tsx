import React, { DetailedHTMLProps, InputHTMLAttributes } from 'react';
import { Color, Size } from '../../types/types';
import './Input.scss';

export type InputType = 'text' | 'email' | 'password';

interface IInputProps
	extends DetailedHTMLProps<
			InputHTMLAttributes<HTMLInputElement>,
			HTMLInputElement
		>,
		React.AriaAttributes {
	label?: string;
	type?: InputType;
	color?: Color;
	scale?: Size;
}

export default React.memo(function Input({
	label,
	type = 'text',
	scale = 'medium',
	className,
	...props
}: IInputProps) {
	const classes = `input input--${scale}${className ? ' ' + className : ''}`;
	return (
		<>
			{label && <label className='input__label'>{label}</label>}
			<input data-testid='input' type={type} className={classes} {...props} />
		</>
	);
});
