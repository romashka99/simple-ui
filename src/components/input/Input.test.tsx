import { render, screen } from '@testing-library/react';
import React from 'react';
import Input from './Input';
import userEvent from '@testing-library/user-event';

describe('Test Input', () => {
	describe('type text', () => {
		test('without attr', () => {
			render(<Input />);
			const input = screen.getByTestId('input');
			expect(input.classList.length).toBe(2);
			expect(input.classList.contains('input')).toBe(true);
			expect(input.classList.contains('input--medium')).toBe(true);
			expect(input.getAttribute('disabled')).toBeNull();
			expect(input).toMatchSnapshot();
		});

		test('size large', () => {
			render(<Input scale='large' />);
			const input = screen.getByTestId('input');
			expect(input.classList.length).toBe(2);
			expect(input.classList.contains('input')).toBe(true);
			expect(input.classList.contains('input--large')).toBe(true);
			expect(input.getAttribute('disabled')).toBeNull();
			expect(input).toMatchSnapshot();
		});

		test('size small', () => {
			render(<Input scale='small' />);
			const input = screen.getByTestId('input');
			expect(input.classList.length).toBe(2);
			expect(input.classList.contains('input')).toBe(true);
			expect(input.classList.contains('input--small')).toBe(true);
			expect(input.getAttribute('disabled')).toBeNull();
			expect(input).toMatchSnapshot();
		});
	});
});
