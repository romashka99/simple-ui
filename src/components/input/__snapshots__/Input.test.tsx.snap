// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input type text size large 1`] = `
<input
  class="input input--large"
  data-testid="input"
  type="text"
/>
`;

exports[`Test Input type text size small 1`] = `
<input
  class="input input--small"
  data-testid="input"
  type="text"
/>
`;

exports[`Test Input type text without attr 1`] = `
<input
  class="input input--medium"
  data-testid="input"
  type="text"
/>
`;
